event/event.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package event is a package that models events that occur during
+// the execution of a program.
+package event
+
+// Event is an interface that custom events should implement.
+// It is empty for now because there are no general methods
+// required of events yet.
+type Event interface {
+}
+
+// KeyCharacter is an event that occurs when a character is actually typed on
+// the keyboard. This may be provided by an input method.
+type KeyCharacter struct {
+	// Code of the key pressed or released.
+	Code int
+	// Key board modifiers.
+	Modifiers int
+	// Character typed.
+	Character rune
+}
+
+// KeyDown is an event that occurs when a key is pressed on the keyboard.
+type KeyDown struct {
+	// Code of the key pressed or released.
+	Code int
+	// Key board modifiers.
+	Modifiers int
+}
+
+// KeyUp is an event that occurs when a key is released on the keyboard.
+// The data is the same as for a KeyDown event.
+type KeyUp KeyDown
+
+// GamepadAxis is for event where an axis on a gamepad changes.
+type GamepadAxis struct {
+	// ID represents which gamepad caused the event.
+	ID int
+	// Axis of the gamepad that changed, if any.
+	Axis int
+	// Position of the axis after the change.
+	// It varies between -1.0 and 1.0.
+	Position float32
+}
+
+// GamepadButtonDown is a gamepad button press event.
+type GamepadButtonDown struct {
+	// ID represents which gamepad caused the event.
+	ID int
+	// Button that was pressed, if any.
+	Button int
+	// Pressure applied to the gamepad button.
+	// It varies between 0.0 for not pressed, and 1.0 for completely pressed.
+	Pressure float32
+}
+
+// GamepadButtonDown is a gamepad button release event.
+// The data is identical to a GamePadButtonDown event.
+type GamepadButtonUp GamepadButtonDown
+
+// GamepadAttach happens when a new gamepad is attached.
+type GamepadAttach struct {
+	// ID represents which gamepad caused the event.
+	ID int
+	// Axes represents the amount of axes the gamepad has.
+	Axes int
+	// Buttons represents the amount of buttons the gamepad has.
+	Buttons int
+}
+
+// GamepadDetach happens when a gamepad is detached.
+type GamepadDetach struct {
+	// ID represents which gamepad caused the event.
+	ID int
+}
+
+// MouseAxes is a mouse axis event.
+type MouseAxis struct {
+	// X position of the mouse pointer.
```

Comment:
The comment needs to be fixed.

Reply:
Ok, but see later for MouseAxis -> MouseMove.